components/lcd/esp_lcd_st7796/esp_lcd_st7796.c

```diff
@@ -0,0 +1,296 @@
+    {0xe1, {0xf0, 0x09, 0x0b, 0x06, 0x04, 0x03, 0x2d, 0x43, 0x42, 0x3b, 0x16, 0x14, 0x17, 0x1b}, 14},
+    {0xf0, {0x3c}, 1},
+    {0xf0, {0x69}, 1},
+    {0, {0}, 0xff},
+};
+
+static esp_err_t panel_st7796_init(esp_lcd_panel_t *panel)
+{
+    st7796_panel_t *st7796 = __containerof(panel, st7796_panel_t, base);
+    esp_lcd_panel_io_handle_t io = st7796->io;
+
+    // LCD goes into sleep mode and display will be turned off after power on reset, exit sleep mode first
+    esp_lcd_panel_io_tx_param(io, LCD_CMD_SLPOUT, NULL, 0);
+    vTaskDelay(pdMS_TO_TICKS(100));
+    esp_lcd_panel_io_tx_param(io, LCD_CMD_MADCTL, (uint8_t[]) {
+        st7796->madctl_val,
+    }, 1);
+    esp_lcd_panel_io_tx_param(io, LCD_CMD_COLMOD, (uint8_t[]) {
+        st7796->colmod_cal,
+    }, 1);
+
+    // vendor specific initialization, it can be different between manufacturers
+    // should consult the LCD supplier for initialization sequence code
+    int cmd = 0;
+    while (vendor_specific_init[cmd].data_bytes != 0xff) {
+        esp_lcd_panel_io_tx_param(io, vendor_specific_init[cmd].cmd, vendor_specific_init[cmd].data, vendor_specific_init[cmd].data_bytes & 0x1F);
+        cmd++;
+    }
+
+    return ESP_OK;
+}
+
+static esp_err_t panel_st7796_draw_bitmap(esp_lcd_panel_t *panel, int x_start, int y_start, int x_end, int y_end, const void *color_data)
+{
+    st7796_panel_t *st7796 = __containerof(panel, st7796_panel_t, base);
+    assert((x_start < x_end) && (y_start < y_end) && "start position must be smaller than end position");
+    esp_lcd_panel_io_handle_t io = st7796->io;
+
+    x_start += st7796->x_gap;
+    x_end += st7796->x_gap;
+    y_start += st7796->y_gap;
+    y_end += st7796->y_gap;
+
+    // define an area of frame memory where MCU can access
+    esp_lcd_panel_io_tx_param(io, LCD_CMD_CASET, (uint8_t[]) {
+        (x_start >> 8) & 0xFF,
+        x_start & 0xFF,
+        ((x_end - 1) >> 8) & 0xFF,
+        (x_end - 1) & 0xFF,
+    }, 4);
+    esp_lcd_panel_io_tx_param(io, LCD_CMD_RASET, (uint8_t[]) {
+        (y_start >> 8) & 0xFF,
+        y_start & 0xFF,
+        ((y_end - 1) >> 8) & 0xFF,
+        (y_end - 1) & 0xFF,
+    }, 4);
+    // transfer frame buffer
+    size_t len = (x_end - x_start) * (y_end - y_start) * st7796->fb_bits_per_pixel / 8;
+    esp_lcd_panel_io_tx_color(io, LCD_CMD_RAMWR, color_data, len);
+
+    return ESP_OK;
+}
+
+static esp_err_t panel_st7796_invert_color(esp_lcd_panel_t *panel, bool invert_color_data)
+{
+    st7796_panel_t *st7796 = __containerof(panel, st7796_panel_t, base);
+    esp_lcd_panel_io_handle_t io = st7796->io;
+    int command = 0;
+    if (invert_color_data) {
+        command = LCD_CMD_INVON;
+    } else {
+        command = LCD_CMD_INVOFF;
+    }
+    esp_lcd_panel_io_tx_param(io, command, NULL, 0);
+    return ESP_OK;
+}
+
+static esp_err_t panel_st7796_mirror(esp_lcd_panel_t *panel, bool mirror_x, bool mirror_y)
```

Comment:
Please, did you test this driver with LVGL rotation? For example this example: `esp-bsp\examples\display_rotation`

Reply:
These commands are part of standard DCS (Display Command Set), and yes, the rotation can work as expect.